source/System.cpp

```diff
@@ -868,6 +897,8 @@ double System::Danger() const
 	double danger = 0.;
 	for(const auto &fleet : fleets)
 	{
+		if(fleet.Period() <= 0)
```

Comment:
```suggestion
		if(!fleet.Period())
```
we dont support negative ones anyway

Reply:
Fleet.Period() is an int, and people can do crazy things in conditions. It isn't a question of whether we support it. The engine shouldn't go crazy if someone provides bad data.

*However*, the RandomEvent does not allow a negative period, and unit tests will check for that, so your implicit bool conversion will work.

Reply:
I mean its fine as it was not like we have to change it

Reply:
I have manually applied this change. Both System.cpp and Engine.cpp now use implicit boolean conversion to test Fleet.Period() instead of a binary comparison.